File: src/test/scala/udemy_2/_5_control_flow_i/_2_other/_6_assignment/_0_intro/Demo7.scala
package udemy_2._5_control_flow_i._2_other._6_assignment._0_intro

object Demo7 extends App {

  var c = new C(0)
  c ?= 1        // c = c ? 1 or c.?(1)
  println(c.value)
}

class C(val value: Int) {
  def ?(k: Int): C = new C(value + k)
}

File: src/test/scala/udemy_2/_5_control_flow_i/_2_other/_6_assignment/_5_calculated/Demo7.java
package udemy_2._5_control_flow_i._2_other._6_assignment._5_calculated;

public class Demo7 {
    public static void main(String[] args) {
        String x = "A";
        String y = "B";

        // read calculated
        char ch = ((x.length() > y.length()) ? x : y).charAt(0);
    }
}
